tests/js/spec/redux/reducers/substanceSearchEntry.spec.jsx

```diff
@@ -39,25 +39,75 @@ describe('substance reducer', () => {
     });
   });
 
-  it('should handle SUBSTANCE_SEARCH_ENTRIES_GET_SUCCESS', () => {
+  it('should handle SUBSTANCE_SEARCH_ENTRIES_GET_SUCCESS for not-grouped', () => {
+    // Arrange
     const prevState = {
       ...initialState,
       loading: true,
       errorMessage: 'oops',
     };
 
-    const nextState = substanceSearchEntry(prevState, {
+    const action = {
       type: 'SUBSTANCE_SEARCH_ENTRIES_GET_SUCCESS',
-      substanceSearchEntries: mockResponseNoGroup,
+      substanceSearchEntries: JSON.parse(JSON.stringify(mockResponseNoGroup)),
```

Comment:
Is there a reason for both stringifying and parsing? Isn't it the same result as `mockResponseNoGroup` on its own?

Reply:
The intent of this is to make a deep copy of mockResponseNoGroup. This was actually a nasty thing! The api response is enhanced in the reducer, and I tried to make a failing test for that, but didn't succeed. It turned out that mockResponseNoGroup was mutated (!) via substanceSearchEntries when sent in to the reducer.

Reply:
I see. Good that you caught that! That's a bug in the reducer then. Could you add a bug ticket for that?

Reply:
The state shouldn't be mutable in the reducer! I'll write a ticken on this, and fix it asap.